pkg/cmd/controller/config/config.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	restclient "k8s.io/client-go/rest"
+)
+
+// Config is configuration struct for EtcdProxyController. It's used to wire CLI and the controller.
+type EtcdProxyController struct {
+	// EtcdProxy contains information about the controller, such as its namespace and etcd proxy image name.
+	EtcdProxy *EtcdProxyOptions
+
+	Kubeconfig *restclient.Config
+}
+
+// CoreEtcdOptions type is used to wire the core etcd information used by controller to create ReplicaSets.
+type CoreEtcdOptions struct {
+	URL             string
+	CAConfigMapName string
+	CertSecretName  string
+}
+
+// EtcdProxyOptions type is used to pass information from cli to the controller.
+type EtcdProxyOptions struct {
```

Comment:
call them config here instead of options

Reply:
Fixed and moved to EtcdProxyController package as proposed by @deads2k.